### PR TITLE
DO NOT MERGE: prototype openshift-tests wrapper/separation

### PR DESCRIPTION
**Pull request by bparees**
goes along with https://github.com/openshift/origin/pull/27563

### CODE REVIEW

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -35,11 +35,21 @@ require (
...
+	github.com/google/gofuzz => github.com/google/gofuzz v1.1.0
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.2.0
 	github.com/imdario/mergo => github.com/imdario/mergo v0.3.6
 	github.com/matttproud/golang_protobuf_extensions => github.com/matttproud/golang_protobuf_extensions v1.0.1
-	github.com/onsi/ginkgo/v2 => github.com/openshift/onsi-ginkgo/v2 v2.0.0-20221005160638-5fa9cd70cd8c
+	github.com/mohae/deepcopy => github.com/mohae/deepcopy v0.0.0-20170603005431-491d3605edfb
+	github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.11.0
+	github.com/onsi/ginkgo/v2 => github.com/openshift/onsi-ginkgo/v2 v2.0.0-20221104153651-0e2187c0d222
+	github.com/openshift/library-go => github.com/openshift/library-go v0.0.0-20221101142545-76526edf66e8
+	github.com/openshift/origin => github.com/bparees/origin v0.0.0-20221207195106-867dedaea002
```
Comment (deads2k) [inline staging/src/k8s.io/api/go.mod:48]: naughty.  This shouldn't be here.

File: staging/src/k8s.io/api/go.mod [hunk 2]
```diff
@@ -35,11 +35,21 @@ require (
 )
 
 replace (
+	github.com/containerd/containerd => github.com/containerd/containerd v1.4.9
```
Comment (deads2k) [inline staging/src/k8s.io/api/go.mod:38]: this stands out as a likely diamond.  What's it for?

File: staging/src/k8s.io/api/go.mod [hunk 3]
```diff
@@ -35,11 +35,21 @@ require (
...
+	github.com/fsnotify/fsnotify => github.com/fsnotify/fsnotify v1.4.9
+	github.com/google/gofuzz => github.com/google/gofuzz v1.1.0
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.2.0
 	github.com/imdario/mergo => github.com/imdario/mergo v0.3.6
 	github.com/matttproud/golang_protobuf_extensions => github.com/matttproud/golang_protobuf_extensions v1.0.1
-	github.com/onsi/ginkgo/v2 => github.com/openshift/onsi-ginkgo/v2 v2.0.0-20221005160638-5fa9cd70cd8c
+	github.com/mohae/deepcopy => github.com/mohae/deepcopy v0.0.0-20170603005431-491d3605edfb
+	github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.11.0
```
Comment (deads2k) [inline staging/src/k8s.io/api/go.mod:45]: v1 ginkgo looks off.

File: staging/src/k8s.io/api/go.mod (hunk 1 above)
Comment (bparees) [reply staging/src/k8s.io/api/go.mod:48]: meaning the origin dep in general?  yeah, ultimately we probably need to factor some code out into a new repo and then have origin + o/k both dep that code.

luckily it is not *that* much code at this point so i don't think it'll be too terrible to do.

**Comment by deads2k** [comment]
some of the deps look off, but I like the idea here.  The wrapper doesn't look awful.  I wonder if we actually need to instantiate all the providers since we're going to execve eventually anyway, right?

**Pull request closed without merging**